### PR TITLE
Misc cropper cleanups

### PR DESCRIPTION
**Pull request by theefer**
A set of code cleanups prior to extending the cropper to support full exports.

This should not introduce any functional change.

See individual commits for a list of changes.

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/model/Export.scala [hunk 1]
```diff
@@ -12,8 +12,7 @@ case class Export(
   date: Option[DateTime],
   specification: CropSource,
   master: Option[Asset],
-  assets: List[Asset],
-  exportType: String
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/model/Export.scala]: This was not used by anything AFAIK.

File: cropper/app/lib/Crops.scala [hunk 2]
```diff
@@ -60,24 +61,25 @@ object Crops {
...
-      val cropTooBig = bounds.width > dimensions.width || bounds.height > dimensions.height
+    asset.dimensions.exists { dimensions =>
+      val positiveCoords       = List(bounds.x,     bounds.y     ).forall(_ >= 0)
+      val strictlyPositiveSize = List(bounds.width, bounds.height).forall(_  > 0)
+      val withinBounds = (bounds.x + bounds.width  <= dimensions.width ) &&
+                         (bounds.y + bounds.height <= dimensions.height)
 
-      outOfBounds || cropTooSmall || cropTooBig
-    } getOrElse true
+      positiveCoords && strictlyPositiveSize && withinBounds
```
Comment (theefer) [inline cropper/app/lib/Crops.scala]: Inverted the logic and made the check more exhaustive (previously, the crop bounds could overflow the actual image dimensions and still be considered valid).

**Comment by kenoir** [comment]
:+1:

File: common-lib/src/main/scala/com/gu/mediaservice/model/Export.scala (hunk 1 above)
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/model/Export.scala]: We do have it in the [Mappings](https://github.com/guardian/grid/blob/master/common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/Mappings.scala#L90) - although I can see it's been moved in your [other PR](https://github.com/guardian/grid/pull/1349/files#diff-fc7552667a451d2b7835c23fa4739ac8R70).

Looking at the ES record though, we are recording it (truncated):
```JSON
{
    "exports": [
        {
            "specification": {},
            "id": "0_161_2447_1467",
            "author": "them.tham@there.co.uk",
            "assets": [
                {},
                {}
            ],
            "date": "2015-06-23T18: 12: 30Z",
            "type": "crop",
            "master": {
                "dimensions": {
                    "height": 1467,
                    "width": 2447
                },
                "secureUrl": ".../0_161_2447_1467/master/2447.jpg",
                "file": .../0_161_2447_1467/master/2447.jpg",
                "mimeType": "image/jpeg",
                "size": 2994287
            }
        }
    ]
}
```
There is also a hard reference to let it be stored in ES [here](https://github.com/guardian/grid/blob/master/cropper/app/controllers/Application.scala#L81).

I don't think it's a problem, just FYI.

Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/model/Export.scala]: Although I see you've remove the `"type" -> "crop"` in the other PR too.

**Pull request merged**